File: CSharp/Guitou/AdventOfCode2022/Solutions/Day1.cs
string inputsPath = "C:\\Users\\guill\\Desktop\\AdventOfCode2022\\CSharp\\Guitou\\AdventOfCode2022\\Inputs\\"; //change to your own

string input = File.ReadAllText(inputsPath + "Input1.txt");

#region Part one
string[] elves = input.Split(Environment.NewLine + Environment.NewLine);
int result = elves.Select(elf =>
{
    string[] calories = elf.Split(Environment.NewLine);
    return calories.Select(x => int.Parse(x)).Sum();
}).Max();

Console.WriteLine(result);
#endregion

#region Part Two
result = elves.Select(elf =>
    { 
        string[] calories = elf.Split(Environment.NewLine); 
        return calories.Select(x => int.Parse(x)).Sum(); 
    })
    .OrderByDescending(x => x)
    .Take(3)
    .Sum();

Console.WriteLine(result);
#endregion

File: Kotlin/ukabi/src/main/kotlin/days1_5/Day1.kt
package days1_5

import inputs.input1

fun day1(input: String, n: Int): Int =
    input
        .split("\n\n")
        .map { sup -> sup.split("\n").sumOf { it.toInt() } }
        .sortedByDescending { it }
        .subList(0, n)
        .sum()

fun main(args: Array<String>) {
    println(day1(input1, 1))
    println(day1(input1, 3))
}